fv3core/stencils/xtp_u.py

```diff
@@ -1,287 +1,171 @@
+
+    Inputs:
+        u: x-dir wind
+        courant: Courant number in flux form
+        rdx: 1.0 / dx
+        bl: ?
+        br: ?
+
+    Returns:
+        flux: Kinetic energy flux
+    """
+    # Could try merging this with xppm version.
+
+    from __externals__ import iord, mord
+
+    b0 = bl + br
+    cfl = courant * rdx[-1, 0, 0] if courant > 0 else courant * rdx
+    fx0 = xppm.fx1_fn(cfl, br, b0, bl)
+
+    if __INLINED(iord < 8):
+        if __INLINED(mord == 5):
+            smt5 = bl * br < 0
+        else:
+            smt5 = (3.0 * abs(b0)) < abs(bl - br)
 
-@gtstencil()
-def get_flux_u_ord8plus(q: sd, c: sd, rdx: sd, bl: sd, br: sd, flux: sd):
-    with computation(PARALLEL), interval(...):
-        b0 = get_b0(bl, br)
-        cfl = c * rdx[-1, 0, 0] if c > 0 else c * rdx
-        fx1 = fx1_fn(cfl, br, b0, bl)
-        flux = q[-1, 0, 0] + fx1 if c > 0.0 else q + fx1
+        if smt5[-1, 0, 0]:
+            tmp = smt5[-1, 0, 0]
+        else:
+            tmp = smt5[-1, 0, 0] + smt5[0, 0, 0]
 
+        flux = xppm.final_flux(courant, u, fx0, tmp)
 
-@gtstencil()
-def br_bl_main(q: sd, al: sd, bl: sd, br: sd):
-    with computation(PARALLEL), interval(...):
-        # TODO: add [0, 0, 0] when gt4py bug is fixed
-        bl = get_bl(al=al, q=q)  # noqa
-        br = get_br(al=al, q=q)  # noqa
+    else:
+        flux = xppm.final_flux(courant, u, fx0, 1.0)
 
+    return flux
 
-@gtstencil()
-def br_bl_corner(br: sd, bl: sd):
-    with computation(PARALLEL), interval(...):
-        bl = 0
-        br = 0
 
+def _compute_stencil(
+    courant: FloatField,
+    u: FloatField,
+    flux: FloatField,
+    dx: FloatField,
+    dxa: FloatField,
+    rdx: FloatField,
+):
+    from __externals__ import i_end, i_start, iord, j_end, j_start, namelist
 
-def zero_br_bl_corners_west(br, bl):
-    grid = spec.grid
-    corner_domain = (2, 1, grid.npz)
-    if grid.sw_corner:
-        br_bl_corner(br, bl, origin=(grid.is_ - 1, grid.js, 0), domain=corner_domain)
-    if grid.nw_corner:
-        br_bl_corner(
-            br, bl, origin=(grid.is_ - 1, grid.je + 1, 0), domain=corner_domain
-        )
+    with computation(PARALLEL), interval(...):
 
+        if __INLINED(iord < 8):
+            al = xppm.compute_al(u, dx)
 
-def zero_br_bl_corners_east(br, bl):
-    grid = spec.grid
-    corner_domain = (2, 1, grid.npz)
-    if grid.se_corner:
-        br_bl_corner(br, bl, origin=(grid.ie, grid.js, 0), domain=corner_domain)
-    if grid.ne_corner:
-        br_bl_corner(br, bl, origin=(grid.ie, grid.je + 1, 0), domain=corner_domain)
+            bl = al[0, 0, 0] - u[0, 0, 0]
+            br = al[1, 0, 0] - u[0, 0, 0]
 
+            # Zero corners
+            with horizontal(
+                region[i_start - 1 : i_start + 1, j_start],
+                region[i_start - 1 : i_start + 1, j_end + 1],
+                region[i_end : i_end + 2, j_start],
+                region[i_end : i_end + 2, j_end + 1],
+            ):
+                bl = 0.0
```

Comment:
This long if else block might be more readable if the cases were written in separate functions

Reply:
Good point. This is mixing low-level code and function calls. I'll give this a refactor over the weekend.